### PR TITLE
Added full report command for Modreport Plugin

### PR DESCRIPTION
**Pull request by tanndlin**
Prob gonna go back and make more comments

### CODE REVIEW

File: src/app/plugins/modreport.plugin.ts [hunk 1]
```diff
@@ -35,17 +35,24 @@ export class ModReportPlugin extends Plugin {
...
+  private async _createReport(message: IMessage, user_handle: string, description?: string) {
+    const id = await Moderation.Helpers.resolveUser(this.container.guildService.get(), user_handle);
+    if (!id) {
+      return;
+    }
+
     const rep: Moderation.Report = new Moderation.Report(
       this.container.guildService.get(),
-      user_handle,
+      id,
```
Comment (c650) [inline src/app/plugins/modreport.plugin.ts:55]: this is that patch right? isnt that checked in already?

File: src/app/plugins/modreport.plugin.ts [hunk 2]
```diff
@@ -67,13 +78,36 @@ export class ModReportPlugin extends Plugin {
     );
   }
 
+  private async _handleFullList(message: IMessage, user_handle: string) {
+    try {
+      await message.reply(`Full Report for ${user_handle}`, {
+        files: [
+          await this.container.modService.getRullReport(
```
Comment (c650) [inline src/app/plugins/modreport.plugin.ts:85]: did you mean `getFullReport` not `getRullReport` or are you scooby doo ?

File: src/services/moderation.service.ts [hunk 3]
```diff
@@ -331,6 +318,110 @@ export class ModService {
     return reply;
   }
 
+  public async getRullReport(guild: Guild, user_handle: string) {
```
Comment (c650) [inline src/services/moderation.service.ts:321]: ruh roh

File: src/services/moderation.service.ts [hunk 4]
```diff
@@ -331,6 +318,110 @@ export class ModService {
...
+
+    const reports = await modreports?.find({ guild: guild.id, user: id }).toArray();
+    const warnings = await modwarnings?.find({ guild: guild.id, user: id }).toArray();
+    const banStatus = await this._getBanStatus(collections, guild, id);
+
+    if (!reports) {
+      throw new Error('Couldnt get reports');
+    }
+
+    //Number of Reports > warns
```
Comment (c650) [inline src/services/moderation.service.ts:339]: `//<space>`

Comment (c650) [reply src/services/moderation.service.ts:339]: does `prettier` not do that for u

File: src/services/moderation.service.ts [hunk 5]
```diff
@@ -331,6 +318,110 @@ export class ModService {
...
+  }
+
+  private _serializeReportForTable(report: Moderation.IModerationReport): string {
+    const ret = `Reported on: ${report.timeStr}<br />Description: ${report.description ||
+      'No Description'}`;
+    if (!report.attachments?.length) {
+      return ret;
+    }
+
+    return `${ret}<br />Attachment: <img src="${report.attachments[0]}">`;
```
Comment (c650) [inline src/services/moderation.service.ts:409]: there can actually be multiple attachments and they might not be images, fyi

yikes that makes it kinda annoying... maybe just see if it ends in .png or .jpg, or just dump all the links

File: src/app/plugins/modreport.plugin.ts (hunk 1 above)
Comment (tanndlin) [reply src/app/plugins/modreport.plugin.ts:55]: I didnt update the dev branch before making this PR, next time i push it should be gone

File: src/services/moderation.service.ts (hunk 4 above)
Comment (tanndlin) [reply src/services/moderation.service.ts:339]: I dont think so :(, lemme see if there's a setting

**Review by c650 — APPROVED** [review]

**Pull request merged**